### ISSUE TITLE
unsuccessful training

### ISSUE THREAD

**Issue by Staniks0** [Pengxin-Guo/FedSA-LoRA#11]
Hi, thank you for your great and inspiring work!

I tried to run your code by your guidance in readme file. However, I think my training is unsuccessful because there is no convergence and a large loss. Some of the latest program output is as following:

high loss problem:

2025-07-07 08:53:44,190 (server:566) INFO: {'Role': 'Client #1', 'Round': 1000, 'Results_raw': {'val_loss': 7102.675781, 'val_total': 6550, 'val_avg_loss': 1.084378, 'val_accuracy': 0.296947}}
2025-07-07 08:53:44,190 (server:566) INFO: {'Role': 'Client #2', 'Round': 1000, 'Results_raw': {'val_loss': 2425.525391, 'val_total': 2154, 'val_avg_loss': 1.126056, 'val_accuracy': 0.538069}}
2025-07-07 08:53:44,191 (server:566) INFO: {'Role': 'Client #3', 'Round': 1000, 'Results_raw': {'val_loss': 1647.94043, 'val_total': 1111, 'val_avg_loss': 1.483295, 'val_accuracy': 0.017102}}

convergence problem:

(monitor:338) INFO: We will compress the file eval_results.raw into a .gz file, and delete the old one
2025-07-07 08:53:44,407 (monitor:246) INFO: After merging the system metrics from all works, we got avg: defaultdict(None, {'id': 'sys_avg', 'sys_avg/fl_end_time_minutes': 669.234474, 'sys_avg/total_model_size': '255.49M', 'sys_avg/total_flops': '0.0', 'sys_avg/total_upload_bytes': '0.0', 'sys_avg/total_download_bytes': '65.99M', 'sys_avg/global_convergence_round': 0.0, 'sys_avg/local_convergence_round': 0.0, 'sys_avg/global_convergence_time_minutes': 0.0, 'sys_avg/local_convergence_time_minutes': 0.0})
2025-07-07 08:53:44,407 (monitor:249) INFO: After merging the system metrics from all works, we got std: defaultdict(None, {'id': 'sys_std', 'sys_std/fl_end_time_minutes': 0.027739, 'sys_std/total_model_size': '147.51M', 'sys_std/total_flops': '0.0', 'sys_std/total_upload_bytes': '0.0', 'sys_std/total_download_bytes': '11.02M', 'sys_std/global_convergence_round': 0.0, 'sys_std/local_convergence_round': 0.0, 'sys_std/global_convergence_time_minutes': 0.0, 'sys_std/local_convergence_time_minutes': 0.0})

Your suggestion would be highly appreciated!

**Comment by Pengxin-Guo** [comment]
Hi,

Since this dataset is sensitive to hyperparameters (as discussed in issue #8), such as learning rate and seed, or possibly GPU devices, you could change these two hyperparameters to give it a try. Thank you.
